WD-Firmware/psp/api_inc/psp_asm_macros.h

```diff
@@ -73,73 +87,157 @@
+    m_STORE    x24, 21  * REGBYTES(sp)
+    m_STORE    x25, 22  * REGBYTES(sp)
+    m_STORE    x26, 23  * REGBYTES(sp)
+    m_STORE    x27, 24  * REGBYTES(sp)
+    m_STORE    x28, 25  * REGBYTES(sp)
+    m_STORE    x29, 26  * REGBYTES(sp)
+    m_STORE    x30, 27  * REGBYTES(sp)
+    m_STORE    x31, 28  * REGBYTES(sp)
 .endm
 
 /* Macro for restoring task context */
 .macro m_popREGFILE
     /* Restore registers,
     Skip global pointer because that does not change */
     m_LOAD    x1,   1 * REGBYTES(sp)
-    m_LOAD    x4,   4 * REGBYTES(sp)
-    m_LOAD    x5,   5 * REGBYTES(sp)
-    m_LOAD    x6,   6 * REGBYTES(sp)
-    m_LOAD    x7,   7 * REGBYTES(sp)
-    m_LOAD    x8,   8 * REGBYTES(sp)
-    m_LOAD    x9,   9 * REGBYTES(sp)
-    m_LOAD    x10, 10 * REGBYTES(sp)
-    m_LOAD    x11, 11 * REGBYTES(sp)
-    m_LOAD    x12, 12 * REGBYTES(sp)
-    m_LOAD    x13, 13 * REGBYTES(sp)
-    m_LOAD    x14, 14 * REGBYTES(sp)
-    m_LOAD    x15, 15 * REGBYTES(sp)
-    m_LOAD    x16, 16 * REGBYTES(sp)
-    m_LOAD    x17, 17 * REGBYTES(sp)
-    m_LOAD    x18, 18 * REGBYTES(sp)
-    m_LOAD    x19, 19 * REGBYTES(sp)
-    m_LOAD    x20, 20 * REGBYTES(sp)
-    m_LOAD    x21, 21 * REGBYTES(sp)
-    m_LOAD    x22, 22 * REGBYTES(sp)
-    m_LOAD    x23, 23 * REGBYTES(sp)
-    m_LOAD    x24, 24 * REGBYTES(sp)
-    m_LOAD    x25, 25 * REGBYTES(sp)
-    m_LOAD    x26, 26 * REGBYTES(sp)
-    m_LOAD    x27, 27 * REGBYTES(sp)
-    m_LOAD    x28, 28 * REGBYTES(sp)
-    m_LOAD    x29, 29 * REGBYTES(sp)
-    m_LOAD    x30, 30 * REGBYTES(sp)
-    m_LOAD    x31, 31 * REGBYTES(sp)
-    m_ADDI    sp, sp, REGBYTES * 36
+    m_LOAD    x5,   2 * REGBYTES(sp)
+    m_LOAD    x6,   3 * REGBYTES(sp)
+    m_LOAD    x7,   4 * REGBYTES(sp)
+    m_LOAD    x8,   5 * REGBYTES(sp)
+    m_LOAD    x9,   6 * REGBYTES(sp)
+    m_LOAD    x10,  7 * REGBYTES(sp)
+    m_LOAD    x11,  8 * REGBYTES(sp)
+    m_LOAD    x12,  9 * REGBYTES(sp)
+    m_LOAD    x13, 10 * REGBYTES(sp)
+    m_LOAD    x14, 11 * REGBYTES(sp)
+    m_LOAD    x15, 12 * REGBYTES(sp)
+    m_LOAD    x16, 13 * REGBYTES(sp)
+    m_LOAD    x17, 14 * REGBYTES(sp)
+    m_LOAD    x18, 15 * REGBYTES(sp)
+    m_LOAD    x19, 16 * REGBYTES(sp)
+    m_LOAD    x20, 17 * REGBYTES(sp)
+    m_LOAD    x21, 18 * REGBYTES(sp)
+    m_LOAD    x22, 19 * REGBYTES(sp)
+    m_LOAD    x23, 20 * REGBYTES(sp)
+    m_LOAD    x24, 21 * REGBYTES(sp)
+    m_LOAD    x25, 22 * REGBYTES(sp)
+    m_LOAD    x26, 23 * REGBYTES(sp)
+    m_LOAD    x27, 24 * REGBYTES(sp)
+    m_LOAD    x28, 25 * REGBYTES(sp)
+    m_LOAD    x29, 26 * REGBYTES(sp)
+    m_LOAD    x30, 27 * REGBYTES(sp)
+    m_LOAD    x31, 28 * REGBYTES(sp)
+    m_ADDI    sp, sp, REGBYTES * 30
+.endm
+
+/* Macro for setting SP to use stack dedicated to ISRs */
+.macro m_SET_SP_TO_ISR_STACK
+    /* Load sp register with the addres of current Task-CB */
+    m_LOAD    sp, xISRStackTop
+.endm
+
+/* Macro for setting SP to use stack of current Task */
+.macro m_SET_SP_TO_TASK_STACK
+    .global    pxCurrentTCB
```

Comment:
1. .global to be out of the macro itself
2. add comment : TODO- add stack check 
3. add input to the macro so no RTOS will be related.
4. change naming to APP_SP_TO_ISR_SP  and ISR _SP_TO APP_SP